src/ssh.c

```diff
@@ -2044,8 +2006,8 @@ static int _UpdateChannelWindow(WOLFSSH_CHANNEL* channel)
 
     inputBuffer = &channel->inputBuffer;
 
-    if (!channel->ssh->isKeying &&
-        (inputBuffer->length > inputBuffer->bufferSz / 2)) {
+    if ((inputBuffer->length > inputBuffer->bufferSz / 2) ||
+         (channel->windowSz == 0)) {
```

Comment:
Changing the default highwater mark for more re-keying fails to send the full file now with the wolfsftp example.


```
$ ./configure CPPFLAGS="-DDEFAULT_HIGHWATER_MARK=16000" --enable-sftp && make

jak@ubuntu:~/Documents/wolfssh$ ls -la large-test.txt
-rw-rw-r-- 1 jak jak 2140640000 Dec 21 10:47 large-test.txt
jak@ubuntu:~/Documents/wolfssh$ ./examples/sftpclient/wolfsftp -u jill
Password: 
wolfSSH sftp> pwd
/home/jak/Documents/wolfssh
wolfSSH sftp> put large-test.txt test.txt
pushing large-test.txt to /home/jak/Documents/wolfssh/test.txt
Processed    32768	 bytes in 0 seconds
wolfSSH sftp> quit
jak@ubuntu:~/Documents/wolfssh$ ls -la test.txt 
-rw-r--r-- 1 jak jak 32768 Feb  1 08:17 test.txt
```

Reply:
This is an issue with the client, not the server. The library wasn't clearing the error register, so a successful return was getting seen as a failure, terminating the connection very early.